integration_test/tests/disconnect_tests.rs

```diff
@@ -0,0 +1,130 @@
+mod common;
+
+mod integration {
+    use crate::common::{
+        allocate_env, allocate_statement, connect_with_conn_string,
+        default_setup_connect_and_alloc_stmt, disconnect_and_free_dbc_and_env_handles,
+        get_sql_diagnostics,
+    };
+    use definitions::{
+        AttrOdbcVersion, FreeStmtOption, Handle, HandleType, SQLExecDirectW, SQLFreeHandle,
+        SQLFreeStmt, SqlReturn, SQL_NTS,
+    };
+
+    #[test]
+    fn sql_disconnect_closes_statement_implicitly() {
+        let (env, dbc, stmt) = default_setup_connect_and_alloc_stmt(AttrOdbcVersion::SQL_OV_ODBC3);
+        let query = b"SELECT * FROM integration_test.foo\0".map(|b| b as u16);
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt, query.as_ptr(), SQL_NTS as i32,),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, *stmt.cast::<Handle>()),
+            );
+
+            // we will not free the statement, instead proceeding directly to disconnect and freeing the dbc and env handles
```

Comment:
```suggestion
            // Directly disconnect and free the dbc and env handles without freeing the statement
```
Small nit about using present tense instead of future tense